src/clj/puppetlabs/puppetserver/jruby_request.clj

```diff
@@ -67,10 +67,10 @@
      (handler (assoc request :jruby-instance jruby-instance)))))
 
 (defn get-environment-from-request
-  "Gets the environment from a request."
+  "Gets the environment from the URL or query string of a request."
   [req]
-  (-> req
-      (get-in [:params "environment"])))
+  (or (get-in req [:params :environment])
+      (get-in req [:params "environment"])))
```

Comment:
Giving priority to the parameter we extract from the comidi route over any that might have been found from a query parameter or form parameter seems reasonable to me.

I think it would be good to be consistent between what this helper method does and how the parameter would be translated onto the request map that we construct and send on down to JRuby.  Prior to this line ...

https://github.com/puppetlabs/puppet-server/blob/puppet-server-2.3.0/src/clj/puppetlabs/services/request_handler/request_handler_core.clj#L278

... in the handling of a request, I think the params map could have both an `:environment` and `environment` key in it.  Doesn't seem like `stringify-keys` would ensure that the keyword form of the key would take precedence over the string one always?  Might depend upon the order in which keys were added to the map?  Might be good to replace this call with something which would explicitly give the keywordized keys predence over the string based ones?

@cprice404 - what do you think?